src/components/layout/Header.js

```diff
@@ -1,28 +1,56 @@
-import React from "react";
-import { NavLink } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+import { Link, NavLink, useLocation } from "react-router-dom";
 
 import classNames from "classnames";
+import useSize from "../../hooks/useSize";
 
 import "./Header.scss";
 
 import logo from "./logo.svg";
 
-const Header = ({ light, sticky }) => (
-  <header className={classNames({ light: light && !sticky, sticky })}>
-    <NavLink to="/" id="header--home-link">
-      <img src={logo} alt="" width="45" />
-      Team Health Check
-    </NavLink>
-    <NavLink activeClassName="header-nav--current" to="/interview/theory">
-      How to ?
-    </NavLink>
-    <NavLink activeClassName="header-nav--current" to="/interview/tryit">
-      Try it
-    </NavLink>
-    <NavLink activeClassName="header-nav--current" to="/visualize">
-      My team
-    </NavLink>
-  </header>
-);
+const Header = ({ light, sticky }) => {
+  const { medium } = useSize();
+  const { pathname } = useLocation();
+
+  const [hidden, setHidden] = useState(medium);
+
+  useEffect(() => {
+    setHidden(medium);
+  }, [medium]);
+
+  useEffect(() => {
+    setHidden(true);
+  }, [pathname]);
```

Comment:
Not sure to understand what you are trying to achieve here... If you change location you display header again ?

Reply:
If you change location you hide the menu